presto-main/src/test/java/io/prestosql/sql/query/TestUnwrapCastInComparison.java

```diff
@@ -214,6 +218,107 @@ public void testVarchar()
         }
     }
 
+    @Test
+    public void testCastTimestampToTimestampWithTimeZone()
+    {
+        ZoneId sessionZone = assertions.getDefaultSession().getTimeZoneKey().getZoneId();
+        verify(sessionZone.getId().equals("Pacific/Apia"), "The values in this test assume Pacific/Apia is the session zone");
```

Comment:
Why not build a session with that time zone instead? Then, the test won't depend on external factors and will be self-contained.

Reply:
I didn't want to overload `validate`, but OK, will do.